### PR TITLE
Issue#74 Network Availability Helper

### LINKED ISSUE #74: Network Availability Helper
It would be nice to have an example of how to verify Network Availability.
This could be as a library for our internal projects, but also as an example to refer back to if ever needed.

An example was already given, modify that to match this repository:
```kotlin
@Suppress("NestedBlockDepth")
class GetNetworkStateUseCaseImpl(private val context: Context) : GetNetworkStateUseCase {
    override fun invoke(): GetNetworkStateUseCase.State {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            connectivityManager.activeNetwork?.let { activeNetwork ->
                connectivityManager.getNetworkCapabilities(activeNetwork)?.let { networkCapabilities ->
                    if (networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
                        && networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)
                    ) {
                        GetNetworkStateUseCase.State.ACTIVE
                    } else {
                        GetNetworkStateUseCase.State.INACTIVE
                    }
                } ?: GetNetworkStateUseCase.State.INACTIVE
            } ?: GetNetworkStateUseCase.State.INACTIVE
        } else {
            @Suppress("DEPRECATION")
            if (connectivityManager.activeNetworkInfo?.isConnected == true) {
                GetNetworkStateUseCase.State.ACTIVE
            } else {
                GetNetworkStateUseCase.State.INACTIVE
            }
        }
    }
}
```

### PR DESCRIPTION
**Pull request by fknives**
Added Network Availability Helper as requested in #74 

Added An example usage into the App module.

**Tested the following scenarios:**
- Turned off WiFi and SIM Internet => showed properly Inactive
- Turned on WiFi with proper connection => showed properly Active
- Turned on SIM Internet with proper connection => showed properly Active
- Turned on WiFi, but set invalid Manual Proxy => showed properly InActive

Created separate module, because it needs `android.permission.ACCESS_NETWORK_STATE` permission.

### CODE REVIEW

File: network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+val ConnectivityManager.isNetworkActiveCompat: Boolean
+    get() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        isNetworkActive
+    } else {
+        isNetworkActiveDeprecated
+    }
+
+@get:RequiresApi(Build.VERSION_CODES.M)
+val ConnectivityManager.isNetworkActive: Boolean
```
Comment (zsoltboldizsar) [inline network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt:34]: It seems to me that we could hide this, what do you think?

File: network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+/**
+ * Checks if the device's active Network has internet capability.
+ *
+ * @return true if Device has Active Internet Available Network, false otherwise.
```
Comment (zsoltboldizsar) [inline network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt:12]: Minor, thing but I would probably rephrase this to: `@return true if device has Available Network and Active Internet, false otherwise.` as it implies both the order and the requirements we're checking. How does that sound to you?

**Review by zsoltboldizsar — CHANGES_REQUESTED** [review]
Great to have these extremely commonly required extensions present in our library for both use and further reference! Thank you :thumbsup:!

File: network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt (hunk 1 above)
Comment (fknives) [reply network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt:34]: My thinking here was that, if your min API is above 23, you wouldn't use the compat + deprecated things anymore, just copy this instead

**Review by zsoltboldizsar — APPROVED** [review]
